File: 03-poo/12-enumeraciones-y-genericos/scala/soluciones/RetoNivelYCaja.scala
enum NivelCurso:
  case Basico, Intermedio, Avanzado
class CajaReto[T]:
  private var valor: Option[T] = None
  def guardar(v: T): Unit = valor = Some(v)
  def obtener: Option[T] = valor

@main def retoNivelYCaja(): Unit =
  val caja = new CajaReto[String]
  caja.guardar("Recurso POO")
  println(s"${NivelCurso.Intermedio} - ${caja.obtener}")

File: 03-poo/12-enumeraciones-y-genericos/java/soluciones/RetoNivelYCaja.java
public class RetoNivelYCaja {
    enum NivelCurso { BASICO, INTERMEDIO, AVANZADO }
    static class Caja<T> { private T valor; void guardar(T valor){this.valor=valor;} T obtener(){return valor;} }
    public static void main(String[] args){ Caja<String> caja=new Caja<>(); caja.guardar("Recurso POO"); System.out.println(NivelCurso.INTERMEDIO+" - "+caja.obtener()); }
}
